### PR TITLE
chore: add workflow to automatically run prettier when renovate updates it

### PR DESCRIPTION
**Pull request by JamesHenry**
## PR Checklist

- [ ] Addresses an existing open issue: fixes #000
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

If a prettier update causes formatting to change (which it usually does) then right now a maintainer has to manually update the relevant renovate PR by running `yarn format` locally and committing the results.

After this PR goes in, that `yarn format` and commit step will be done automatically on the PR without any maintainer involvement

### CODE REVIEW

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
It would have been nice to get this natively in Renovate with [`postUpgradeTasks`](https://docs.renovatebot.com/configuration-options/#postupgradetasks) but it says they're only supported in self-hosted instances 😞...

```json
{
  "postUpgradeTasks": {
    "commands": ["yarn format"],
    "fileFilters": ["**/*"]
  }
}
```

But, this feels like something a lot of repos have to solve in more situations. Surely folks need this whenever they update Prettier? So it's something I think we should tackle more holistically.

What about a post-commit hook to run `yarn format` when the root `package.json` changes? [[their closest example, ignoring filename changes](https://github.com/okonet/lint-staged#example-run-tsc-on-changes-to-typescript-files-but-do-not-pass-any-filename-arguments)]

**Comment by JamesHenry** [comment]
@JoshuaKGoldberg would a local hook be triggered/included in renovate's commit though?

**Comment by JoshuaKGoldberg** [comment]
Hmm, good question. I don't actually know. Never mind me 😄

File: .github/workflows/prettier-update.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          git diff HEAD~1 -G"prettier" --exit-code package.json && echo "prettier unchanged" || echo "::set-output name=was-changed::true"
+
+      - name: Run prettier formatting if prettier was changed and commit the results
+        if: ${{ steps.prettier-package-check.outputs.was-changed == 'true' }}
+        env:
+          # We cannot use secrets.GITHUB_TOKEN for this because it is not permitted to kick off subsequent actions worfklow runs, so we use a PAT instead
+          GITHUB_TOKEN: ${{ secrets.JAMES_HENRY_GITHUB_TOKEN }}
+        run: |
+          yarn --ignore-scripts
+          yarn format
```
Comment (JoshuaKGoldberg) [inline .github/workflows/prettier-update.yml:37]: For a bit faster install, maybe?

```suggestion
          npx prettier --write .
```

**Review by JoshuaKGoldberg — APPROVED** [review]
:shipit: Thanks!

File: .github/workflows/prettier-update.yml (hunk 1 above)
Comment (bradzacher) [reply .github/workflows/prettier-update.yml:37]: will `npx` respect the `package.json` version of prettier?

Comment (JoshuaKGoldberg) [reply .github/workflows/prettier-update.yml:37]: It goes with the latest by default, which should be what the PR is upgrading to anyway.

File: .github/workflows/prettier-update.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+          # Commit all the changes to the PR (see note on not being able to use secrets.GITHUB_TOKEN for this)
+          git config --global user.email "james@henry.sc"
+          git config --global user.name "JamesHenry"
+          git remote set-url origin https://x-access-token:$GITHUB_TOKEN@github.com/$GITHUB_REPOSITORY.git
+
+          git add --all
+          git commit -m "chore: update formatting after prettier upgrade"
```
Comment (fisker) [inline .github/workflows/prettier-update.yml:45]: Will this fail if nothing changed.

Comment (JamesHenry) [reply .github/workflows/prettier-update.yml]: Good catch, thanks @fisker

**Pull request merged**